clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -0,0 +1,219 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * A unit test for KafkaAdminClient.
+ *
+ * See KafkaAdminClientIntegrationTest for an integration test of the KafkaAdminClient.
+ */
+public class KafkaAdminClientTest {
+    @Rule
+    public Timeout globalTimeout = Timeout.millis(120000);
+
+    @Test
+    public void testGetOrCreateListValue() {
+        Map<String, List<String>> map = new HashMap<>();
+        List<String> fooList = KafkaAdminClient.getOrCreateListValue(map, "foo");
+        assertNotNull(fooList);
+        fooList.add("a");
+        fooList.add("b");
+        List<String> fooList2 = KafkaAdminClient.getOrCreateListValue(map, "foo");
+        assertEquals(fooList, fooList2);
+        assertTrue(fooList2.contains("a"));
+        assertTrue(fooList2.contains("b"));
+        List<String> barList = KafkaAdminClient.getOrCreateListValue(map, "bar");
+        assertNotNull(barList);
+        assertTrue(barList.isEmpty());
+    }
+
+    @Test
+    public void testGetTimeoutMsRemainingAsInt() {
+        assertEquals(0, KafkaAdminClient.getTimeoutMsRemainingAsInt(1000, 1000));
+        assertEquals(100, KafkaAdminClient.getTimeoutMsRemainingAsInt(1000, 1100));
+        assertEquals(Integer.MAX_VALUE, KafkaAdminClient.getTimeoutMsRemainingAsInt(0, Long.MAX_VALUE));
+        assertEquals(Integer.MIN_VALUE, KafkaAdminClient.getTimeoutMsRemainingAsInt(Long.MAX_VALUE, 0));
+    }
+
+    @Test
+    public void testPrettyPrintException() {
+        assertEquals("Null exception.", KafkaAdminClient.prettyPrintException(null));
+        assertEquals("TimeoutException", KafkaAdminClient.prettyPrintException(new TimeoutException()));
+        assertEquals("The foobar timed out.",
+            KafkaAdminClient.prettyPrintException(new TimeoutException("The foobar timed out.")));
+    }
+
+    @Test
+    public void testCloseQuietly() {
+        KafkaAdminClient.closeQuietly("nothing", null);
+        KafkaAdminClient.closeQuietly("ByteArrayOutputStream", new ByteArrayOutputStream());
+        KafkaAdminClient.closeQuietly("ErrorCloser", new AutoCloseable() {
+            @Override
+            public void close() throws Exception {
+                throw new KafkaException("oh no");
+            }
+        });
+    }
+
+    private static Map<String, Object> newStrMap(String... vals) {
+        Map<String, Object> map = new HashMap<>();
+        map.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:8121");
+        map.put(AdminClientConfig.REQUEST_TIMEOUT_MS_CONFIG, "1000");
+        if (vals.length % 2 != 0) {
```

Comment:
If we want `vals` to always have two values, then we should just make the method take two parameters?